.github/workflows/ci.yaml

```diff
@@ -0,0 +1,1144 @@
+        # use for the live docs site in that we only test a single version of
+        # the content.
+        #
+        # To do this, we delete the three submodules we use for building the
+        # live docs site and copy a gravitational/teleport clone into the
+        # content directory.
+        #
+        # The docs engine expects a config.json file at the root of the
+        # gravitational/docs clone that associates directories with git
+        # submodules. By default, these directories represent versioned branches
+        # of gravitational/teleport. We override this in order to build only a
+        # single version of the docs.
+        #
+        # We also replace data fetched from Sanity CMS with hardcoded JSON
+        # objects to remove the need to authenticate with Sanity. Each includes
+        # the minimal set of data required for docs builds to succeed.
+        run: |
+          echo "" > .gitmodules
+          rm -rf content/*
+          # Rather than using a submodule, copy the teleport source into the
+          # content directory.
+          cp -r "$GITHUB_WORKSPACE/teleport" "$GITHUB_WORKSPACE/docs/content/current"
+          jq -nr --arg version "current" '{"versions": [{"name": $version,"branch": $version,"deprecated": false,"isDefault": true}]}' > config.json
+          NEW_PACKAGE_JSON=$(jq '.scripts."git-update" = "echo Skipping submodule update"' package.json);
+          NEW_PACKAGE_JSON=$(jq '.scripts."prepare-sanity-data" = "echo Using pre-populated Sanity data"' <<< "$NEW_PACKAGE_JSON");
+          echo "$NEW_PACKAGE_JSON" > package.json;
+          echo "{}" > data/events.json
+          echo '{"bannerButtons":{"second":{"title":"LOG IN","url":"https://teleport.sh"},"first":{"title":"Support","url":"https://goteleport.com/support/"}},"navbarData":{"rightSide":{},"logo":"/favicon.svg","menu":[]}}' > data/navbar.json
+
+      - name: Check spelling
+        working-directory: 'docs'
+        run: yarn spellcheck content/current
+
+      - name: Lint docs formatting
+        working-directory: 'docs'
+        run: yarn markdown-lint
+
+      - name: Test the docs build
+        working-directory: docs
+        run: yarn build
+
+  build-api:
+    name: Build API
+    needs: changes
+    if: ${{ needs.changes.outputs.has_api == 'true' }}
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout Teleport
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          # use the version declared in API's go.mod
+          go-version-file: api/go.mod
+          cache-dependency-path: api/go.sum
+
+      - name: Build
+        run: cd api; go build ./...
+
+  build-macos:
+    name: Build on Mac OS
+    needs: changes
+    if: ${{ !startsWith(github.head_ref, 'dependabot/') && (needs.changes.outputs.has_go == 'true' || needs.changes.outputs.has_rust == 'true') && github.event_name == 'merge_group' }}
```

Comment:
This and related `if`s match non-Teleport Go changes (such as tooling Go changes)